de.dlr.sc.virsat.model.extension.fdir.ui/src/de/dlr/sc/virsat/model/extension/fdir/ui/diagram/ra/features/states/StateMoveFeature.java

```diff
@@ -27,37 +33,62 @@
  */
 public class StateMoveFeature extends VirSatMoveShapeFeature {
 
+	private static final String Y_POS_REL_TO_FAULT_NODE = "y-pos-rel-to-fault-node";
+	private static final String X_POS_REL_TO_FAULT_NODE = "x-pos-rel-to-fault-node";
+
 	/**
 	 * Standard constructor
 	 * @param fp the feature provider
 	 */
 	public StateMoveFeature(IFeatureProvider fp) {
 		super(fp);
 	}
-	
+
 	@Override
 	public void moveShape(IMoveShapeContext context) {
 		ContainerShape containerShape = (ContainerShape) context.getShape();
-		
+
+		PictogramElement pe = context.getPictogramElement();
+		State bean = (State) getBusinessObjectForPictogramElement(pe);
+
 		Connection targetConnection = context.getTargetConnection();
 		if (targetConnection != null) {
 			Anchor start = targetConnection.getStart();
 			Anchor end = targetConnection.getEnd();
-			
+
 			Anchor movedObjectAnchor = containerShape.getAnchors().get(0);
-			
+
 			ReconnectionContext reconnectionContext = new ReconnectionContext(targetConnection, start, movedObjectAnchor, null);
 			reconnectionContext.setReconnectType(ReconnectionContext.RECONNECT_TARGET);
 			getFeatureProvider().getReconnectionFeature(reconnectionContext).reconnect(reconnectionContext);
-			
+
 			CreateConnectionContext createConnectionContext = new CreateConnectionContext();
 			createConnectionContext.setSourceAnchor(movedObjectAnchor);
 			createConnectionContext.setTargetAnchor(end);
 			new FaultEventTransitionCreateFeature(getFeatureProvider()).create(createConnectionContext);
 
 		}
-		
+
 		super.moveShape(context);
+		moveComments(containerShape, bean);
 	}
-	
+
+	/**
+	 * @param containerShape to layout comments for
+	 * @param bean business object of pictogram
+	 */
+	private void moveComments(ContainerShape containerShape, State bean) {

```

Comment:
We probably have some common code here. Can we maybe move the moveComments to the CommentUtil and is it in both fault tree node and recovery automaton state layouting?